src/Features/Core/Portable/FindUsages/DefinitionItem.cs

```diff
@@ -109,6 +111,8 @@ internal abstract partial class DefinitionItem
 
         internal abstract bool IsExternal { get; }
 
+        protected readonly Workspace Workspace;
```

Comment:
💭 Why are we storing the workspace here?

Reply:
will doc.  this is necessary for things like MetadataAsSource.  Say we are in a metadata-file (for a symbol like IDisposable) and we ask for all the impls of it.  We'll have DocSpans pointing into the normal VSWorkspace.  in order to then navigate to these later, we need *that* workspace (the VSworkspace) to find the docs in, not the workspace hte goto-impl *command* originates in (the MAS workspace).

Previously this just worked as we stored DocSpans (which point at this workspace).  If we're only storing doc-ids, we need some mechanism to make this work.  Happy to hear thoughts if there's a better way to do this.

Reply:
Works for me thanks